### PR TITLE
squiid 1.0.6 (new formula)

### PR DESCRIPTION
**Pull request by TabulateJarl8**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by TabulateJarl8** [comment]
Does anyone know what this error means? I kept getting it on my machine and couldn't figure it out

**Comment by chenrui333** [comment]
it is brew audit issue.

File: Formula/s/squiid.rb
Comment (chenrui333) [inline Formula/s/squiid.rb:16]: checkout the other terminal app about the test part.

I am curious why `ctrl+c` or `ctrl+q` are not supported with the app?

**Comment by chenrui333** [comment]
you can also squash the commits using the commit msg as the PR title `squiid 1.0.6 (new formula)`

**Comment by TabulateJarl8** [comment]
> it is brew audit issue.

Do I have to do anything then? Should I start an issue?

File: Formula/s/squiid.rb
Comment (TabulateJarl8) [reply Formula/s/squiid.rb:16]: For some reason I don't quite understand this. If you're asking why ctrl+c and ctrl+q aren't supposed, it's because it's a TUI app that captures the terminal, and we don't handle those key combinations

**Comment by TabulateJarl8** [comment]
Could I get a review on this? Thanks

File: Formula/s/squiid.rb
Comment (chenrui333) [reply Formula/s/squiid.rb:16]: you can find other examples on the TUI tests, like terminal navigator stuff.

**Comment by chenrui333** [comment]
still need a bit work on the TUI test, other than that LGTM

**Comment by TabulateJarl8** [comment]
I don't have a device that runs macOS so it'll take a while for me to get a VM set up since it works on Linux but not macOS. I'll push a commit or update when I figure out the issue

**Comment by TabulateJarl8** [comment]
Hope those tests are enough, should be ready

File: Formula/s/squiid.rb
Comment (ZhongRuoyu) [inline Formula/s/squiid.rb:65]: There is too much verbosity in the test. Can you simplify it? Something like:

```suggestion
    PTY.spawn("squiid") do |r, w, pid|
      sleep 1 # wait for squiid to start

      w.write "(10 - 2) * (3 + 5) / 4\r"
      assert_match "(10-2)*(3+5)/4=16", read_stdout(r)
      w.write "quit\r"

      # for some reason the test hangs on macOS until stdout is read again
      read_stdout(r) unless OS.linux?
      Process.wait(pid)
    rescue PTY::ChildExited
      # app exited
    end
```

Comment (TabulateJarl8) [reply Formula/s/squiid.rb:65]: Seems to work on my end

File: Formula/s/squiid.rb
Comment (ZhongRuoyu) [inline Formula/s/squiid.rb:3]: It seems that the project homepage is at https://imaginaryinfinity.net/projects/squiid/ precisely.

File: Formula/s/squiid.rb
Comment (ZhongRuoyu) [inline Formula/s/squiid.rb:8]: [The dependency on the `nng-sys` crate](https://gitlab.com/ImaginaryInfinity/squiid-calculator/squiid/-/blob/1.0.6/squiid-engine/Cargo.toml?ref_type=tags#L13) is likely the reason why CMake is needed. However, vendored dependencies are generally discouraged, and we do have an `nng` formula.

I'll soon push a change to force linking with brewed `nng`.

File: Formula/s/squiid.rb
Comment (ZhongRuoyu) [inline Formula/s/squiid.rb:6]: According to https://gitlab.com/ImaginaryInfinity/squiid-calculator/squiid it's `GPL-3.0-or-later`.

**Review by ZhongRuoyu — COMMENTED** [review]
These comments will be addressed in my next push. Adding them for clarity.

**Comment by TabulateJarl8** [comment]
Thank you for those changes!

**Review by ZhongRuoyu — APPROVED** [review]
Thank you, @TabulateJarl8.

**Comment by TabulateJarl8** [comment]
@ZhongRuoyu quick question, can I do anything to speed up the process of this merge? Is it just waiting on another review?

**Review by p-linnane — APPROVED** [review]

**Comment by ZhongRuoyu** [comment]
Was just waiting on another review; thanks for being patient. Thanks @p-linnane for the review too.

**Pull request merged**
